File: Client/projects/fibi/src/app/admin-modules/person-training/person-training.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { CommonService } from '../../common/services/common.service';
import { PreviousSearchObj } from './person-training.interface';

@Injectable()
export class PersonTrainingService {
    previousSearch = new PreviousSearchObj();
    formData = new FormData();

    constructor(private _http: HttpClient, private _commonService: CommonService) {
    }

    loadPersonTrainingList(params) {
        return this._http.post(this._commonService.baseUrl + '/getTrainingDashboard', params);
    }

    getPersonTrainingDetails(trainingId: number) {
        return this._http.get(`${this._commonService.baseUrl}/getPersonTrainingDetails/${trainingId}`);
    }

    downloadTrainingAttachment(trainingAttachmentId: number) {
        return this._http.get(this._commonService.baseUrl + '/downloadtrainingAttachment/' + trainingAttachmentId,
            {observe: 'response', responseType: 'blob'});
    }

    saveOrUpdatePersonTraining(params: any) {
        return this._http.post(this._commonService.baseUrl + '/saveOrUpdatePersonTraining', params);
    }

    saveOrUpdateTrainingComments(params) {
        return this._http.post(this._commonService.baseUrl + '/saveOrUpdateTrainingComments', params);
    }

    deleteTrainingComments(trainingCommentId: number) {
        return this._http.delete(this._commonService.baseUrl + '/deleteTrainingComments/' + trainingCommentId);
    }

    deletePersonTraining(personTrainingId: number) {
        return this._http.delete(this._commonService.baseUrl + '/deletePersonTraining/' + personTrainingId);
    }

    deleteTrainingAttachment(trainingAttachmentId: number) {
        return this._http.delete(this._commonService.baseUrl + '/deleteTrainingAttachment/' + trainingAttachmentId);
    }

    saveOrUpdateTrainingAttachment(uploadedFile, requestObject) {
        this.formData.delete('fileData');
        this.formData.delete('formDataJson');
        this.formData.append('fileData', uploadedFile);
        this.formData.append('formDataJson', JSON.stringify(requestObject));
        return this._http.post(this._commonService.baseUrl + '/saveOrUpdateTrainingAttachment', this.formData);
    }
}

File: Server/FCOI/src/main/java/com/polus/fibicomp/persontraining/service/PersonTrainingService.java
package com.polus.fibicomp.persontraining.service;

import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;
import org.springframework.web.multipart.MultipartFile;

import com.polus.fibicomp.person.vo.PersonVO;

@Service
public interface PersonTrainingService {

	/**
	 * @param vo
	 * @return updated person training
	 */
	public String saveOrUpdatePersonTraining(PersonVO vo);

	/**
	 * @param vo
	 * @return training list
	 */
	public String getTrainingDashboard(PersonVO vo);

	/**
	 * @param vo
	 * @return updated comments
	 */
	public String saveOrUpdateTrainingComments(PersonVO vo);

	/**
	 * @param trainingCommentId
	 * @return boolean
	 */
	public String deleteTrainingComments(Integer trainingCommentId);

	/**
	 * @param trainingId
	 * @return training details
	 */
	public String getPersonTrainingDetails(Integer trainingId);

	/**
	 * @param searchString
	 * @return vo
	 */
	public String loadTrainingList(String searchString);

	/**
	 * @param files
	 * @param formDataJson
	 * @return vo
	 */
	public String saveOrUpdateTrainingAttachment(MultipartFile[] files, String formDataJson);

	/**
	 * @param parseInt
	 * @return blob
	 */
	public ResponseEntity<byte[]> downloadtrainingAttachment(int parseInt);

	/**
	 * @param attachmentId
	 * @return boolean
	 */
	public String deleteTrainingAttachment(Integer attachmentId);

	/**
	 * @param personTrainingId
	 * @return boolean
	 */
	public String deletePersonTraining(Integer personTrainingId);

}
